lexorus/foursquare-restaurants-challenge
# Refactor FoursquareVenueFetcher to contain only fetching logic

This issue resolves #1 

### Background
I took a shortcut because of the limited time and didn't implement the `FoursquareVenueFetcher` in the best way. It needs some love, but since it's encapsulated and covered with tests, it shouldn't be a problem.

### What has been done
1. Each request will have its own structure that should inherit from
`FoursquareRequest`
2. For building URLRequest from `FoursquareRequest` will be used
`FoursquareRequestFactory`
**Note:** The refactoring of `FoursquareVenueFetcher` didn't affect its public interface in any way. So no changes have been done to the application code, just to the `VenuesAPI` library internals.

### How to test
1. Since there is no CI yet, check that tests are passing.
2. The application functionality shouldn't be affected after this change at all.
